README.md

```diff
@@ -1,2 +1,115 @@
-# KokkosP Profiling Tools
-Performance profiling tools that can be optionally integrated into Kokkos-based applications and libraries.
+
+
+# Kokkos Tools
+
+Kokkos Tools provide a light weight set of profiling and debug utilities, which interface with instrumentation hooks built directly into the Kokkos runtime. Compared to 3rd party tools these tools can provide much cleaner, context specific information: in particular, they allow kernel centric analysis and they use labels provided to Kokkos objects (kernel launches and views).
+
+Under most circumstances the profiling hooks are compiled into Kokkos executables by default. That means the tools work for your existing Kokkos applications, assuming that the profiling hooks version is compatible with the tools version. No recompilation or changes to your build procedures are required.
+
+Note: `Kokkos_ENABLE_LIBDL` must be on to load profiling hooks dynamically. It should be on by default however.
+
+## General Usage
+
+ To use one of the tools you have to compile it, which will generate a dynamic library. Before executing the Kokkos application you then have to set the environment variable `KOKKOS_PROFILE_LIBRARY` to point to the dynamic libary e.g. in Bash:
+```
+export KOKKOS_PROFILE_LIBRARY=${HOME}/kokkos-tools/src/tools/memory-events/kp_memory_event.so
+```
+
+Many of the tools will produce an output file which uses the hostname as well as the process id as part of the filename. 
+
+## Explicit Instrumentation
+
+One can explicitly add instrumentation to a library or an application. Currently the only hooks intended for explicit programmer use are the Region related hooks. These use a push/pop model to mark coarser regions in your code.
```

Comment:
Did you mean to capitalize the "R" in region?
We also have sections even though most tools don't actually register a callback for those.

Reply:
This could be confusing. I think we ought to explicitly define what a region means in the context of Kokkos profiling tools.

Reply:
(And yes, I did mean to capitalize the R in region.)